falcon/cluster/cluster.py

```diff
@@ -272,191 +358,97 @@ def _dist_mz_interval(
-    precursor_tol_mass : float
-        The precursor tolerance mass for vectors to be considered as neighbors.
-    precursor_tol_mode : str
-        The unit of the precursor m/z tolerance ('Da' or 'ppm').
-    rt_tol : float
-        The retention time tolerance for vectors to be considered as neighbors.
-        If `None`, do not filter neighbors on retention time.
-    nn_dists : np.ndarray
-        Distances of the nearest neighbors.
-    nn_idx_ann : np.ndarray
-        Indexes of the nearest neighbors.
-    n_neighbors : int
-        The (maximum) number of neighbors to set for each vector.
-    distances : np.ndarray
-        The nearest neighbor distances. See `scipy.sparse.csr_matrix` (`data`).
-    indices : np.ndarray
-        The column indices for the nearest neighbor distances. See
-        `scipy.sparse.csr_matrix`.
-    indptr : np.ndarray
-        The index pointers for the nearest neighbor distances. See
-        `scipy.sparse.csr_matrix`.
-    indptr_i : int
-        The current start index in `indptr`.
-    """
-    nn_idx_mz = _get_neighbors_idx(
-        precursor_mzs, batch_start, batch_stop,
-        precursor_tol_mass, precursor_tol_mode)
-    if rt_tol is not None:
-        nn_idx_rt = _get_neighbors_idx(rts, batch_start, batch_stop, rt_tol,
-                                       'rt')
-        nn_idx_mz = [_intersect_idx_ann_mz(idx_mz, idx_rt, None, True)
-                     for idx_mz, idx_rt in zip(nn_idx_mz, nn_idx_rt)]
-    indptr_i_start = indptr_i - batch_start
-    for i, (idx_ann, idx_mz, dists) in enumerate(
-            zip(nn_idx_ann, nn_idx_mz, nn_dists), indptr_i):
-        mask = _intersect_idx_ann_mz(idx_ann, idx_mz, n_neighbors)
-        indptr[i + 1] = indptr[i] + len(mask)
-        # Convert cosine similarity to cosine distance.
-        distances[indptr[i]:indptr[i + 1]] = np.maximum(1 - dists[mask], 0)
-        indices[indptr[i]:indptr[i + 1]] = indptr_i_start + idx_ann[mask]
-
-
-@nb.njit(cache=True)
-def _get_neighbors_idx(values: np.ndarray, start_i: int, stop_i: int,
-                       tol: float, tol_mode: str) \
-        -> List[np.ndarray]:
+        for i, vec in enumerate(vectors, batch_start):
+            sel = faiss.IDSelectorBatch(nn_mz[i % batch_size])
+            params = faiss.SearchParametersIVF(sel=sel, nprobe=1)
+            # noinspection PyArgumentList
+            vec = vec.reshape(1, -1)
+            d, idx = index.search(vec, n_neighbors, params=params)
+            # filter out indices and distances < 0
+            mask = idx >= 0
+            d, idx = d[mask], idx[mask]
+            # Build csr matrix
+            indptr[i + 1] = indptr[i] + len(d)  # d.shape[1]  # len(d)
+            # Convert cosine similarity to cosine distance.
+            distances[indptr[i] : indptr[i + 1]] = np.maximum(1 - d, 0)
+            indices[indptr[i] : indptr[i + 1]] = idx
+        batch_start = batch_stop
+
+
+def _get_mz_neighbors(
```

Comment:
**todo:** The function name doesn't seem appropriate anymore, because it also covers filtering neighbors on RT now.